src/producer.c

```diff
@@ -2,15 +2,36 @@
 
 /**
  *
- * Producer thread will read from the file and write data to 
+ * Producer thread will read from the file and write data to
  * the end of the shared queue
  */
 void *producer(void *arg){
-    
+
+    printf("producer\n");
+
     //TODO: open the file and read its content line by line
-    //Send data to the shared queue
+
+    FILE * fd_in = (FILE*)arg; //  file we are reading from; will be passed in
+    ssize_t nread;
+    char buffer[chunkSize]; // chunk size for now
```

Comment:
`getLineFromFile` uses `getline`

I just read this in official c documentation:

   ```
        ssize_t getline(char **restrict lineptr, size_t *restrict n, FILE *restrict stream);
        
        .....
        
        If *lineptr is set to NULL before the call, then getline() will
       allocate a buffer for storing the line.  This buffer should be
       freed by the user program even if getline() failed.

       Alternatively, before calling getline(), *lineptr can contain a
       pointer to a malloc(3)-allocated buffer *n bytes in size.  If the
       buffer is not large enough to hold the line, getline() resizes it
       with realloc(3), updating *lineptr and *n as necessary.

       In either case, on a successful call, *lineptr and *n will be
       updated to reflect the buffer address and allocated size
       respectively.
```

So if what you did doesn't work consider just passing a pointer like: 
`char* buffer = NULL;` 
and then `free` it at the end.